client/components/tinymce/plugins/media/drop-zone.jsx

```diff
@@ -107,18 +144,67 @@ export default React.createClass( {
 		if ( isSingleImage && ! MediaValidationStore.hasErrors( site.ID ) ) {
 			// For single image upload, insert into post content, blocking save
 			// until the image has finished upload
-			if ( selectedItems[ 0 ].transient ) {
+			let selectedItem = selectedItems[ 0 ]
+			if ( selectedItem.transient ) {
 				PostActions.blockSave( 'MEDIA_MODAL_TRANSIENT_INSERT' );
 			}
 
-			onInsertMedia( markup.get( selectedItems[ 0 ] ) );
-			MediaActions.setLibrarySelectedItems( site.ID, [] );
-		} else {
-			// In all other cases, show the media modal list view
-			onRenderModal( { visible: true } );
+			analytics.mc.bumpStat( 'editor_upload_via', 'editor_drop' );
+
+			if ( config.isEnabled( 'post-editor/image-reorientation' ) ) {
+				let handleImageOrientation = ( orientation ) => {
+					// If the orientation is `1` (i.e., "top"), then no further orientation
+					// changes are necessary.
+					if ( orientation > 1 ) {
```

Comment:
Similar to previous note about early return, I think you could do the same here:

```js
if ( 1 === orientation ) {
	return Promise.resolve( selectedItem );
}
```